crates/pallet-bioauth/src/lib.rs

```diff
@@ -390,18 +400,29 @@ pub mod pallet {
                 .any(|authentication| &authentication.public_key == public_key)
         }
 
-        pub fn deauthenticate(public_key: &<T as Config>::ValidatorPublicKey) -> bool {
-            ActiveAuthentications::<T>::mutate(|active_authentications| {
+        pub fn deauthenticate(
+            public_keys: Vec<<T as Config>::ValidatorPublicKey>,
+            reason: <T as Config>::DeauthenticationReason,
+        ) -> bool {
+            let removed = ActiveAuthentications::<T>::mutate(|active_authentications| {
                 let mut removed = false;
                 active_authentications.retain(|authentication| {
-                    if &authentication.public_key == public_key {
+                    if public_keys.contains(&authentication.public_key) {
                         removed = true;
                         return false;
                     }
                     true
                 });
                 removed
-            })
+            });
+            if removed {
+                // Emit an event.
+                Self::deposit_event(Event::AuthenticationsRemoved {
+                    removed: public_keys,
+                    reason,
+                });
+            }
+            removed
         }
```

Comment:
This implementation does not have the ability to accurately communicate whether a public key was actually removed from the set or not. As in, the resulting value also must be an array of public keys instead of being a simple bool.

One of the properties of the old version was that it was possible to invoke `deauthenticate` and get a `false`, which would make sense to log to check where those non-actionable deauthentications come from and how many of them are usually in the chain.